src/main/java/baubles/client/gui/BaublesGuiFactory.java

```diff
@@ -1,27 +1,41 @@
 package baubles.client.gui;
 
-import java.util.Set;
-import net.minecraft.client.Minecraft;
+import java.util.ArrayList;
+import java.util.List;
 import net.minecraft.client.gui.GuiScreen;
-import net.minecraftforge.fml.client.IModGuiFactory;
+import net.minecraftforge.common.config.ConfigElement;
+import net.minecraftforge.common.config.Configuration;
+import net.minecraftforge.fml.client.DefaultGuiFactory;
+import net.minecraftforge.fml.client.config.GuiConfig;
+import net.minecraftforge.fml.client.config.IConfigElement;
+import baubles.common.Baubles;
+import baubles.common.Config;
 
-public class BaublesGuiFactory implements IModGuiFactory {
+public class BaublesGuiFactory extends DefaultGuiFactory {
 
-	@Override
-	public void initialize(Minecraft minecraftInstance) { }
+	public BaublesGuiFactory() {
+		super(Baubles.MODID, getTitle());
```

Comment:
You can just put getAbridgedConfigPath here, and then call the field "title" in createConfigGui. Same thing with modid really.
Example: https://github.com/squeek502/WailaHarvestability/blob/1.12.x/java/squeek/wailaharvestability/gui/config/GuiFactory.java